src/CattyTests/Views/Data/BrickCellVariableDataTests.swift

```diff
@@ -0,0 +1,120 @@
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+import XCTest
+
+@testable import Pocket_Code
+
+class BrickCellVariableDataTests: XCTestCase {
+
+    var project: Project!
+    var spriteObject: SpriteObject!
+    var script: Script!
+    var brick: ChangeVariableBrick!
+    var brickCell: ChangeVariableBrickCell!
+
+    var variablesContainer: VariablesContainer!
+    var objectVariable1: UserVariable!
+    var objectVariable2: UserVariable!
+    var programVariable: UserVariable!
+
+    override func setUp() {
+
+        spriteObject = SpriteObject()
+        spriteObject.name = "testObject"
+
+        variablesContainer = VariablesContainer()
+        objectVariable1 = UserVariable(name: "testVariable1")
+        objectVariable2 = UserVariable(name: "testVariable2")
+        programVariable = UserVariable(name: "testVariable3")
+
+        variablesContainer.addObjectVariable(objectVariable1, for: spriteObject)
+        variablesContainer.addObjectVariable(objectVariable2, for: spriteObject)
+        variablesContainer.programVariableList.add(programVariable as Any)
+
+        project = Project()
+        project.variables = variablesContainer
+
+        spriteObject.project = project
+
+        script = Script()
+        script.object = spriteObject
+
+        brick = ChangeVariableBrick()
+        brick.script = script
+        brickCell = ChangeVariableBrickCell()
+
+    }
+
+    func testInitWithFrameWhenObjectVariableInContainer() {
+
+        brick.userVariable = objectVariable2
+        brickCell.scriptOrBrick = brick
+
+        let brickCellVariableData = BrickCellVariableData(frame: CGRect(), andBrickCell: brickCell, andLineNumber: 0, andParameterNumber: 0)
+
+        XCTAssertTrue(brickCellVariableData?.currentValue == brick.userVariable.name)
+
+        let values = brickCellVariableData?.values as? [String]
+        XCTAssertEqual(values?[0], kLocalizedNewElement)
+        XCTAssertEqual(values?[1], programVariable.name)
+        XCTAssertEqual(values?[2], objectVariable1.name)
+        XCTAssertEqual(values?[3], objectVariable2.name)
+    }
+
+    func testInitWithFrameWhenProgramVariableInContainer() {
+
+       brick.userVariable = programVariable
+       brickCell.scriptOrBrick = brick
+
+        let brickCellVariableData = BrickCellVariableData(frame: CGRect(), andBrickCell: brickCell, andLineNumber: 0, andParameterNumber: 0)
+
+        XCTAssertTrue(brickCellVariableData?.currentValue == brick.userVariable.name)
+
+        let values = brickCellVariableData?.values as? [String]
+        XCTAssertEqual(values?[0], kLocalizedNewElement)
+        XCTAssertEqual(values?[1], programVariable.name)
+        XCTAssertEqual(values?[2], objectVariable1.name)
+        XCTAssertEqual(values?[3], objectVariable2.name)
+
+    }
+
+    func testInitWithFrameWhenVariableNotInContainer() {
```

Comment:
Thanks for your changes! Please remove this method, since this case should never happen in reality and thus should also not be documented. But there is one scenario missing: please add a test with a second SpriteObject containing one (local) variable and make sure that this variable is not included in the list. General suggestion: Replace `testInitWithFrame` with `testValues`.

Reply:
Please review the latest commit. Here, Instead of adding a separate test I tested that scenario by making sure the count remains 4 and compared each element. If I add a separate method for the same, I will require to declare a new VariablesContainer or add a new local variable of spriteObject2 in the existing VariablesContainer instance inside the new method and testing it in a similar way as in other tests. So I thought this to be a better way. Please let me know if this will be okay.

Reply:
Adding the comparison with _4_ is a good idea. Also the test seems to be alright. In general I would suggest to test one logical behavior in one single test. This makes debugging much easier in case of a test failure. But since these tests are not too complex, that is alright.

However, please also add one test verifying the behavior when no variables are available. Also: I do not understand why you named the methods _WhenObjectVariableInContainer_ and _WhenProgramVariableInContainer_. As far as I can see both methods have the same variables in the container but different variables in the Brick.